Yosemite/Yosemite/PointOfSale/POSProductProvider.swift

```diff
@@ -39,17 +39,15 @@ public final class POSProductProvider: POSItemProvider {
 
             return mapProductsToPOSItems(products: filteredProducts)
         } catch {
-            // TODO:
-            // - Handle case for empty product list, or not empty but no eligible products
-            // https://github.com/woocommerce/woocommerce-ios/issues/12815
-            // https://github.com/woocommerce/woocommerce-ios/issues/12816
-            // - Handle case for error when fetching products
-            // https://github.com/woocommerce/woocommerce-ios/issues/12846
-            DDLogError("No POS items")
-            return []
+            DDLogError("Failed to retrieve products. Error: \(error)")
+            throw POSError.requestFailed
         }
     }
 
+    enum POSError: Error {
+        case requestFailed
+    }
```

Comment:
I went ahead with a simple structure for the error and a single case, but perhaps we want to define these further and use a common POSError everywhere? We could also pass the error messages as an associated value.

Reply:
Good question. I don't know if we want to or can have a single `POSError`, especially given most of the errors come from payment-related code.

My suggestion would be to have:

a) `POSProductProviderError`

```
enum POSProductProviderError: Error {
        case requestFailed
    }
```

b) Propagate errors from `productsRemote` for now, especially since we don't handle them in any special way

```
    public func providePointOfSaleItems() async throws -> [POSItem] {
        let products = try await productsRemote.loadAllSimpleProductsForPointOfSale(for: siteID)

        let eligibilityCriteria: [(Product) -> Bool] = [
            isNotVirtual,
            isNotDownloadable,
            hasPrice
        ]
        let filteredProducts = filterProducts(products: products, using: eligibilityCriteria)

        return mapProductsToPOSItems(products: filteredProducts)
    }
```

Reply:
Thanks! I'm inclined to go with `a` at the moment, so we're more explicit about the type of error. Updated the naming on https://github.com/woocommerce/woocommerce-ios/pull/13586/commits/2fef97ca97056bf1e7ae29909be5796ad6f305c0